pandas/_libs/src/datetime/np_datetime_strings.c

```diff
@@ -253,9 +253,7 @@ int parse_iso_8601_datetime(char *str, int len,
 
     /* Next character must be a ':' or the end of the string */
     if (sublen == 0) {
```

Comment:
I suppose you don't even need the brackets if this is all that remains within the conditional